engine/src/flutter/common/input/text_input_connection.h

```diff
@@ -0,0 +1,44 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_COMMON_INPUT_TEXT_INPUT_CONNECTION_H_
+#define FLUTTER_COMMON_INPUT_TEXT_INPUT_CONNECTION_H_
+
+#include <string>
+#include "fml/closure.h"
+
+namespace flutter {
+
+class TextInputConnection {
+ public:
+  TextInputConnection() = default;
+
+  virtual ~TextInputConnection() {}
+
+  virtual std::string GetCurrentText() = 0;
+
+  virtual void SetCurrentText(std::string_view text) = 0;
+
+  virtual void SetUpdateCallback(fml::closure callback) {}
+
+  TextInputConnection(const TextInputConnection&) = delete;
+  TextInputConnection& operator=(const TextInputConnection&) = delete;
+};
+
+class TextInputConnectionFactory {
+ public:
+  TextInputConnectionFactory() = default;
+
+  virtual ~TextInputConnectionFactory() {}
+
+  virtual std::shared_ptr<TextInputConnection> CreateTextInputConnection() = 0;
```

Comment:
I assume we would need to provide implementation that convert the call into embedder.h api?

Reply:
Yeah you could have a generic EmbedderTextInputConnectionFactory that would be backed by callbacks provided in embedder.h